### PR TITLE
update to poms so module can be included in repository-list

### PR DESCRIPTION
**Pull request by tsurdilo**

### CODE REVIEW

**Comment by tsurdilo** [comment]
@mswiderski @psiroky Here are the changes @psiroky and I talked about we should do to include jbpm-work-items module in the repository-list.txt:

* added parent 
<parent>
    <groupId>org.kie</groupId>
    <artifactId>kie-parent</artifactId>
    <version>7.5.0-SNAPSHOT</version>
  </parent>

* applying all inherited checks (none have been disabled)
* removed all versions that are already defined in parent pom (just keeping the 3rd party ones not defined there until we can inspect them)
* cleaned up the <build> in all modules
* added depends to all modules so that transitive depends check passes
* updated code to follow the stylecheck checks

I hope this is OK now. Please guys let me know if there is anything else you would like to see changed. We would really like to include this by the end of the month. Thanks!

File: pom.xml [hunk 1]
```diff
@@ -266,47 +214,110 @@
...
+      </plugins>
+    </pluginManagement>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.5.1</version>
+        <configuration>
+          <source>${maven.compiler.source}</source>
+          <target>${maven.compiler.target}</target>
```
Comment (psiroky) [inline pom.xml]: Please remove the version, source and target. These are already coming from kie-parent.

File: pom.xml [hunk 2]
```diff
@@ -266,47 +214,110 @@
...
+          </annotationProcessors>
+          <compilerArgs>
+            <arg>-AwidName=${project.artifactId}.wid</arg>
+          </compilerArgs>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-assembly-plugin</artifactId>
+        <version>3.0.0</version>
```
Comment (psiroky) [inline pom.xml]: Please remove the version (the plugin version is already defined in kie-parent).

**Comment by tsurdilo** [comment]
@psiroky updated. thanks!

File: pom.xml [hunk 3]
```diff
@@ -266,47 +214,108 @@
...
+        </plugin>
+      </plugins>
+    </pluginManagement>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+          <source>${maven.compiler.source}</source>
+          <target>${maven.compiler.target}</target>
```
Comment (psiroky) [inline pom.xml]: These should still be removed. The source+target are set globally in kie-parent.

**Comment by tsurdilo** [comment]
@psiroky done. thanks again!

**Pull request merged**